### ISSUE TITLE
GitHub build statuses don't seem to refresh

### ISSUE THREAD

**Issue by ScottG489** [ScottG489/dashboard#1]
I noticed just now that the build statuses should have updated (from red to green) for a few builds but they didn't when hitting the refresh button. The did update however when I refreshed the page.

**Comment by ScottG489** [comment]
As of 1818ade3508adc80304c742401206a28033dc1f4 this still isn't working.

My hunch why this is happening is that since the json list of badges returned isn't any different, then the component doesn't rerender.

If this is the case, to have it always rerender we could just do something slightly hacky like setting a timestamp field.

**Comment by ScottG489** [comment]
It actually seems like it might rerender when the setter is called regardless:
https://stackoverflow.com/a/24719289/14146969

However, it might be an issue with the DOM not updating?
https://stackoverflow.com/a/24719294/14146969

**Comment by ScottG489** [comment]
On close inspection it seems like the first link actually has the answer at the bottom. Though it initially led me to believe it wouldn't update. It's actually the virtual DOM that updates every time but only the real one if something changes.

>Native DOM renders: React changes real DOM nodes in your browser only if they were changed in the Virtual DOM and as little as needed - this is that great React's feature which optimizes real DOM mutation and makes React fast.

This SO answer lead me to finding out the answer though
https://stackoverflow.com/questions/47922687/force-react-to-reload-an-image-file

I'll have to have something which changes so that it will rerender. Using the current date isn't ideal since it will change every time. Perhaps there's a field I can use that indicates the state of the build? That should change when the build status changes

**Comment by ScottG489** [comment]
I've updated the back-end job to send the status of the build (failed/success). I've taken the status and appended it as an empty query param to the github badge image.

I believe this is the most "proper" solution. Still seems weird that we have to create a new URL in order to keep the browser from caching the image. Seems like there should be a cleaner solution.

However, compared to what I found to be the most common solution of adding a random param to the end this seems cleaner. With that solution it causes the browser to download and cache every single one of those images separately. In my case there are only 2 different images and they correspond to the build status. So by keying off of the build status this will cause the browser to cache both the red and green badge images just once.

Fixed by 4c1fec18be5f7abbc99623b5a650485a841df8bb.
